lib/validation.js

```diff
@@ -28,10 +28,14 @@ function build (context, compile, schemas) {
 
   context.schema = schemas.resolveRefs(context.schema)
 
-  if (context.schema.headers) {
+  const headers = context.schema.headers
+
+  if (headers && headers.constructor !== Object) {
```

Comment:
headers.constructor is a function isn't it?

I think this should be `headers.constructor.name !== 'Object'`

Reply:
Fixed in https://github.com/fastify/fastify/pull/1179.

Reply:
It will be changed to with #1179 
`if (headers && Object.getPrototypeOf(headers) !== Object.prototype) {`